compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5058,20 +4970,26 @@ genericLoadHelper(TR::Node * node, TR::CodeGenerator * cg, TR::MemoryReference *
       TR::InstOpCode::Mnemonic load = loadInstrs[form][numberOfBytesLog2][isSourceSigned][numberOfExtendBits/32-1];
       if (form == RegReg)
          {
-         generateRRInstruction(cg, load, node, targetRegister, srcRegister);
+         if (load == TR::InstOpCode::LR)
```

Comment:
Are we able to make this check more generic and use the instruction formats already stored in the OMRInstOpCodeProperties.hpp tables? i.e. we can check if the instruction is RR format rather than checking a specific opcode. Let's also add an assert in the fallback that the instruction is in fact  an RRE format instruction.

Reply:
Fixed to use format checking e.g. `getInstructionFormat() == RX_FORMAT`.

The `loadInstrs` array only contains two formats for `RegReg` load and two formats for `MemReg` load. So no need to add assert for this.